grpc-interface-support/src/cpp.py

```diff
@@ -346,28 +360,57 @@ def __create_or_update_service_header(self) -> None:
         service_header_file_path = os.path.join(
             app_source_dir, service_header_file_name
         )
-
-        user_defined_code: List[str] = []
-        if os.path.exists(service_header_file_path):
-            # there is a previous version of the header
-            # extract the user-defined code
-
-            user_defined_code = capture_area_in_file(
-                open(service_header_file_path, encoding="utf-8"),
-                "// <user-defined>",
-                "// </user-defined>",
-            )
-
         variables = self.__get_template_variables()
         variables["service_header_code"] = "\n".join(header_stub_code)
-        variables["service_header_user_code"] = "\n".join(user_defined_code)
 
         copy_templates(
             get_template_dir(),
             app_source_dir,
             [CopySpec("ServiceImpl.h", service_header_file_name)],
             variables,
         )
+        
+    def __update_service_header(self) -> None:
+        header_generated_code = GrpcCodeExtractor(
+            self.__proto_file_handle, self.__package_directory_path
+        ).get_header_stub_code(self.__get_include_dir())
+
+        header_generated_code = self.__transform_header_stub_code(header_stub_code)
+
+        app_source_dir = os.path.join(get_workspace_dir(), "app", "src")
+        service_header_file_name = (
+            f"{self.__proto_file_handle.get_service_name()}ServiceImpl.h"
+        )
+        service_header_file_path = os.path.join(
+            app_source_dir, service_header_file_name
+        )
+
+        auto_generated_code = capture_area_in_file(
+            open(service_header_file_path, encoding="utf-8"),
+            "// <auto-generated>",
+            "// </auto-generated>",
+        )
+
+        header_file_content = self.__read_file_to_string(service_header_file_path)
+        modified_content = header_file_content.replace('\n'.join(auto_generated_code), '\n'.join(header_generated_code))
+        self.__write_string_to_file(service_header_file_path, modified_content)
+
+    def __read_file_to_string(self, filepath):
```

Comment:
String manipulation was extracted from this file to the Velocitas lib e.g.
It may make sense to also do this for file manipulation.

```
from velocitas_lib.text_utils import (
    capture_area_in_file,
    to_camel_case,
)
```

Reply:
PR can be found here: https://github.com/eclipse-velocitas/velocitas-lib/pull/23